src/App.tsx

```diff
@@ -1,222 +1,107 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { FilterCondition } from './types/FilterCondition';
+import { Header } from './components/Main/Header';
+import { TodoList } from './components/Main/TodoList';
+import { Footer } from './components/Main/Footer';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+  const [filterCondition, setFilterCondition] = useState(FilterCondition.ALL);
+
+  if (isError) {
+    setTimeout(() => setIsError(false), 3000);
+  }
+
+  const getSelected = (allTodos: Todo[]): Todo[] => {
+    return allTodos.filter(item => {
+      switch (filterCondition) {
+        case FilterCondition.ALL:
+          return true;
+
+        case FilterCondition.COMPLETED:
+          return item.completed === true;
+
+        case FilterCondition.ACTIVE:
+          return item.completed === false;
+
+        default: return true;
+      }
+    });
```

Comment:
Better to check filter type before filtering. In this case you can avoid iteration for `FilterCondition.ALL`

```suggestion
    switch (filterCondition) {
        case FilterCondition.COMPLETED:
          return allTodos.filter(item => (
            item.completed
          )
          
        case FilterCondition.ACTIVE:
          return allTodos.filter(item => (
            !item.completed
          )
        
        case FilterCondition.ALL:
        default:
          return allTodos;
```